labelbox/data/metrics/iou.py

```diff
@@ -1,281 +1,254 @@
-    elif tool_type in NDClassification.get_union_types():
-        return classification_miou(predictions, labels)
-    else:
-        raise ValueError(f"Unexpected annotation found. Found {tool_type}")
+    classification_iou = get_iou_across_features(ground_truth_classifications,
+                                                 prediction_classifications)
+    agreement = _mask_iou(ground_truth_np, prediction_np)
+    classification_iou = classification_iou if classification_iou is not None else agreement
+    return (agreement + classification_iou) / 2.
 
 
-def datarow_miou(label_content: List[Dict[str, Any]],
-                 ndjsons: List[Dict[str, Any]],
-                 include_classifications=True,
-                 include_subclasses=True) -> float:
+def classification_miou(ground_truths: List[ClassificationAnnotation],
+                        predictions: List[ClassificationAnnotation]) -> float:
     """
+    Computes iou score for all features with the same feature schema id.
 
     Args:
-        label_content : one row from the bulk label export - `project.export_labels()`
-        ndjsons: Model predictions in the ndjson format specified here (https://docs.labelbox.com/data-model/en/index-en#annotations)
-        include_classifications: Whether or not to factor top level classifications into the iou score.
-        include_subclassifications: Whether or not to factor in subclassifications into the iou score
+        ground_truths: List of ground truth classification annotations
+        predictions: List of prediction classification annotations
     Returns:
-        float indicating the iou score for this data row.
-
+        float representing the iou score for the classification
     """
 
-    predictions, labels, feature_schemas = _preprocess_args(
-        label_content, ndjsons, include_classifications)
+    if len(predictions) != len(ground_truths) != 1:
+        return 0.
 
-    ious = [
-        feature_miou(predictions[feature_schema],
-                     labels[feature_schema],
-                     include_subclasses=include_subclasses)
-        for feature_schema in feature_schemas
-    ]
-    ious = [iou for iou in ious if iou is not None]
-    if not ious:
-        return None
-    return np.mean(ious)
+    prediction, ground_truth = predictions[0], ground_truths[0]
+
+    if type(prediction) != type(ground_truth):
+        raise TypeError(
+            "Classification features must be the same type to compute agreement. "
+            f"Found `{type(prediction)}` and `{type(ground_truth)}`")
+
+    if isinstance(prediction.value, Text):
+        return text_iou(ground_truth.value, prediction.value)
+    elif isinstance(prediction.value, Radio):
+        return radio_iou(ground_truth.value, prediction.value)
+    elif isinstance(prediction.value, Checklist):
+        return checklist_iou(ground_truth.value, prediction.value)
+    else:
+        raise ValueError(f"Unexpected subclass. {prediction}")
```

Comment:
So dropdown is not currently supported? (because of MAL, I'm guessing?)

Reply:
Yes